### DISCUSSION TITLE
Horizontal Baseline Alignment for Multiple Text Objects

### DISCUSSION THREAD

**Discussion by eiLLFire** [fabricjs/fabric.js#10213] [Q&A]
I need the ability to select two or more text objects and align them horizontally based on their individual baselines. This feature would ensure that text objects with different font sizes, line heights, or font families can be aligned along a common baseline, ensuring a uniform appearance in the layout. How do I implement function <code>getBaselinePosition</code>
![image](https://github.com/user-attachments/assets/6c8a3b12-5494-4418-8248-c2d75635dad0)
```js
var text1 = new Text('Hello', {
  left: 100,
  top: 100,
  fontSize: 40,
});

var text2 = new Text('World', {
  fontFamliy: 'xxx',
  left: 200,
  top: 150,
  fontSize: 60,
});

canvas.add(text1, text2);

function getBaselinePosition(text) {
  // calculate its baseline based on its font metrics
}

var baseline1 = getBaselinePosition(text1);
var baseline2 = getBaselinePosition(text2);
if (baseline1 > baseline2) {
  text2.top += baseline1 - baseline2;
} else {
  text1.top += baseline2 - baseline1;
}
```

**Comment by eiLLFire** [comment]
![image](https://github.com/user-attachments/assets/8fe16fcf-0afb-45b8-87b6-97a20c912511)

**Comment by asturur** [comment]
I do not think we have this feature, but we should expose a method like this, to get the baseline point for lines of text objects

**Comment by eiLLFire** [comment]
Is there a plan to add this method for getting the baseline point of lines in text objects in the next version of v6?

**Comment by asturur** [comment]
There is not a PLAN with a deadline, i just marked as a feature we should have.
